File: ecma6/euler003.js
/**
 * Created by gcharles on 3/23/16.
 */
//
//  The prime factors of 13195 are 5, 7, 13 and 29.
//
//  What is the largest prime factor of the number 600851475143 ?
//
function euler003() {
   const number = 600851475143;

   let factorMap = primes.primeFactors(number);
   return Math.max.apply(null, [...factorMap.factors()]);
}

File: java8/src/Euler003.java
import framework.EulerProblem;
import primes.FactorMap;

import java.util.Comparator;

/**
 * The prime factors of 13195 are 5, 7, 13 and 29.
 * What is the largest prime factor of the number 600851475143 ?
 *
 * Created by gcharles on 8/17/16.
 */
public class Euler003 extends EulerProblem<Long> {
    @Override
    public long solve( Long input ) {
        FactorMap fm = FactorMap.factor( input );
        return fm.factors().stream().max(Comparator.naturalOrder()).get();
    }
}